src/screens/DiscoveryScreenV2/elements/Search.js

```diff
@@ -130,18 +129,12 @@ const DiscoverySearch = ({
     AsyncStorage.setItem(RECENT_SEARCH_TERMS, JSON.stringify(resultArray));
   };
 
-  React.useEffect(() => {
-    setIsSearchIconShown(!isFocus && !isTextAvailable);
-  }, [isTextAvailable, isFocus]);
-
   React.useEffect(() => {
     debounceChangeText(searchText);
-    setIsTextAvailable(searchText.length > 0);
   }, [searchText]);
 
   React.useEffect(() => {
     const unsubscribe = () => {
-      setIsTextAvailable(false);
       setSearchText('');
       DiscoveryAction.setDiscoveryData(
         {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The state variable `isSearchIconShown` and its related logic have been removed. If this state was used elsewhere in the component or had side effects, make sure those are addressed.
```diff
- setIsSearchIconShown(!isFocus && !isTextAvailable);
- setIsTextAvailable(searchText.length > 0);
- setIsTextAvailable(false);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->